### ISSUE TITLE
Error handling js escape when parsing html

### ISSUE THREAD

**Issue by in64** [WebReflection/linkedom#128]
test case on deno

`deno run -A d.js`

d.js
```
import { parseHTML } from 'https://raw.githubusercontent.com/WebReflection/linkedom/main/worker.js';
const r = await (await fetch('https://www.xiaohongshu.com/web-login/canvas?redirectPath=http%3A%2F%2Fwww.xiaohongshu.com', {headers:{'User-Agent': 'ua'}})).text()
const { document } = parseHTML(r)
// left
console.log(r)
// right
console.log(document.toString())
```

<img width="992" alt="image" src="https://user-images.githubusercontent.com/87516105/163176371-a08e9894-0062-46a9-9c35-99fce5adddf2.png">

Thanks!

**Comment by WebReflection** [comment]
```js
const {document: scripted} = parseHTML(
  `<!doctype html><html><head><script> !$ && (o = f["on" + n]); </script></head></html>`
);

assert(scripted.toString(), '<!DOCTYPE html><html><head><script> !$ && (o = f["on" + n]); </script></head></html>');
```

So far, so good ... can you find out what is failing? Because this difference makes very little sense to me, and there are no special escapes or chars in that specific part.

TL;DR can't reproduce, can't work on it.

**Comment by WebReflection** [comment]
FYI: I'll close this in 10 days if, in a month, nothing happened 👋

**Comment by WebReflection** [comment]
Nothing happened
